applications/RANSApplication/custom_processes/rans_check_vector_bounds_process.cpp

```diff
@@ -0,0 +1,186 @@
+#include "utilities/parallel_utilities.h"
+
+// Include base h
+#include "rans_check_vector_bounds_process.h"
+
+namespace Kratos
+{
+RansCheckVectorBoundsProcess::RansCheckVectorBoundsProcess(
+    Model& rModel,
+    Parameters rParameters)
+: mrModel(rModel), mrParameters(rParameters)
+{
+    KRATOS_TRY
+
+    Parameters default_parameters = Parameters(R"(
+        {
+            "model_part_name" : "PLEASE_SPECIFY_MODEL_PART_NAME",
+            "variable_name"   : "PLEASE_SPECIFY_SCALAR_VARIABLE",
+            "component_type"  : "magnitude"
+        })");
+
+    mrParameters.ValidateAndAssignDefaults(default_parameters);
+
+    mVariableName = mrParameters["variable_name"].GetString();
+    mModelPartName = mrParameters["model_part_name"].GetString();
+
+    if (mrParameters["component_type"].GetString() == "magnitude") {
+        mVectorComponent = VectorComponent::Magnitude;
+    } else if (mrParameters["component_type"].GetString() == "x") {
+        mVectorComponent = VectorComponent::X;
+    } else if (mrParameters["component_type"].GetString() == "y") {
+        mVectorComponent = VectorComponent::Y;
+    } else if (mrParameters["component_type"].GetString() == "z") {
+        mVectorComponent = VectorComponent::Z;
+    } else {
+        KRATOS_ERROR
+            << "Vector component_type type not found. [ component_type = "
+            << mrParameters["component_type"].GetString() << " ]\n.";
+    }
+
+    KRATOS_CATCH("");
+}
+
+int RansCheckVectorBoundsProcess::Check()
+{
+    KRATOS_TRY
+
+    const Variable<array_1d<double, 3>>& vector_variable =
+        KratosComponents<Variable<array_1d<double, 3>>>::Get(mVariableName);
+
+    const auto& r_model_part = mrModel.GetModelPart(mModelPartName);
+
+    KRATOS_ERROR_IF(!r_model_part.HasNodalSolutionStepVariable(vector_variable))
+        << mVariableName << " is not found in nodal solution step variables list of "
+        << mModelPartName << ".";
+
+    return 0;
+
+    KRATOS_CATCH("");
+}
+
+void RansCheckVectorBoundsProcess::Execute()
+{
+    KRATOS_TRY
+    const Variable<array_1d<double, 3>>& r_vector_variable =
+        KratosComponents<Variable<array_1d<double, 3>>>::Get(mVariableName);
+
+    const auto& r_model_part = mrModel.GetModelPart(mModelPartName);
+    const auto& r_nodes = r_model_part.Nodes();
+
+    array_1d<double, 3> vector_weights;
+
+    switch (mVectorComponent) {
+    case VectorComponent::Magnitude:
+        vector_weights[0] = 1.0;
+        vector_weights[1] = 1.0;
+        vector_weights[2] = 1.0;
+        break;
+    case VectorComponent::X:
+        vector_weights[0] = 1.0;
+        vector_weights[1] = 0.0;
+        vector_weights[2] = 0.0;
+        break;
+    case VectorComponent::Y:
+        vector_weights[0] = 0.0;
+        vector_weights[1] = 1.0;
+        vector_weights[2] = 0.0;
+        break;
+    case VectorComponent::Z:
+        vector_weights[0] = 0.0;
+        vector_weights[1] = 0.0;
+        vector_weights[2] = 1.0;
```

Comment:
Couldn't we achieve the same result using the `double` version with `COMPONENT_X`, `COMPONENT_Y` and `COMPONENT_Z` variables.

Reply:
removed. We don't need it since as you suggest we have components which are again type of Variable<double> :)